client/src/main/java/io/getstream/chat/android/client/models/Attachment.kt

```diff
@@ -30,6 +32,14 @@ data class Attachment(
     var name: String? = null,
     var fallback: String? = null,
 
+    @IgnoreSerialisation
+    @IgnoreDeserialisation
+    var upload: File? = null,
+
+    @IgnoreSerialisation
+    @IgnoreDeserialisation
+    var uploadError: ChatError? = null,
```

Comment:
Considering we want to mark status of upload I'd prefer some abstraction over `var uploadError: ChatError? = null`. For example:
```kotlin
sealed class UploadState()
object InProgress: UploadState()
object Success: UploadState()
data class Failed(val error: ChatError): UploadState()
```

Reply:
How about adding it to livedata's `MessageEntity` only, keeping this outside LLC?

Reply:
`UploadState` is something that should belong to "Offline-Store" implementation, it is not a valid state in LLC.
On the other hand, our models should be simpler. What "uploadError" mean for someone that is only using LLC?

Reply:
The upload state is indeed somewhat easier to work with, good suggestion.

@JcMinarro about your suggestion, I see your point, what alternative solution do you recommend? I like how easy approach is to use for people integrating with Stream

Reply:
It should be handled on the Offline-Support SDK, MessageEntity, that is stored in our DB, is the good candidate to handle it.